src/cli/usb.js

```diff
@@ -125,6 +125,29 @@ module.exports = ({ commandProcessor, root }) => {
 		}
 	});
 
+	commandProcessor.createCommand(usb, 'cloud-status', 'Check a device\'s cloud connection state', {
+		params: '<deviceID>',
+		options: {
+			'until': {
+				description: 'Poll your device for a specific connection state and then exit',
+				choices: [
+					'unknown',
+					'disconnected',
+					'connecting',
+					'connected',
+					'disconnecting'
+				]
+			}
```

Comment:
🤔 add optional `--timeout` flag and enforce a timeout when `--until` flag is set..?

Reply:
✅ in https://github.com/particle-iot/particle-cli/pull/577/commits/94b64b3051ef95a79cd3224d2879f66dc3477e15